File: lib/cfr/builtins/libs/dart/ui/windowPadding.dart
import 'dart:core';
import 'dart:ui';

import 'package:hydro_sdk/cfr/runtimeSupport.dart';

class VMManagedWindowPadding extends VMManagedBox<WindowPadding> {
  VMManagedWindowPadding(
      {required this.table, required this.vmObject, required this.hydroState})
      : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) {
    table['left'] = vmObject.left;
    table['top'] = vmObject.top;
    table['right'] = vmObject.right;
    table['bottom'] = vmObject.bottom;
    table['toString'] =
        makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
      return [
        vmObject.toString(),
      ];
    });
  }

  final HydroTable table;

  final HydroState hydroState;

  final WindowPadding vmObject;
}

void loadWindowPadding(
    {required HydroState hydroState, required HydroTable table}) {
  table['windowPaddingZero'] =
      makeLuaDartFunc(func: (List<dynamic> luaCallerArguments) {
    return [
      maybeBoxObject<WindowPadding>(
          object: WindowPadding.zero,
          hydroState: hydroState,
          table: HydroTable())
    ];
  });
  registerBoxer<WindowPadding>(boxer: (
      {required WindowPadding vmObject,
      required HydroState hydroState,
      required HydroTable table}) {
    return VMManagedWindowPadding(
        vmObject: vmObject, hydroState: hydroState, table: table);
  });
}

File: runtime/dart/ui/windowPadding.ts
declare const dart: {
    ui: {
        windowPaddingZero: (this: void) => IWindowPadding;
    };
};
export interface IWindowPadding {
    left: number;
    top: number;
    right: number;
    bottom: number;
    toString: () => string;
}
export class WindowPadding {
    public static zero = dart.ui.windowPaddingZero();
    public readonly left: number = undefined as any;
    public readonly top: number = undefined as any;
    public readonly right: number = undefined as any;
    public readonly bottom: number = undefined as any;
    private readonly _dart_toString: () => string = undefined as any;
    public toString(): string {
        return this._dart_toString();
    }
}
